cover.c

```diff
@@ -174,7 +266,8 @@ static void cover_node_ensure(CoverNode* node, int line) {
     /* GLOG(("Growing map for [%s] from %d to %d - %p", node->file, node->alen, size, node->lines)); */
 
     /* realloc will grow the data and keep all current values... */
-    GMEM_REALLOC(node->lines, unsigned char*, node->alen * sizeof(unsigned char*), size * sizeof(unsigned char*));
+    GMEM_REALLOC(node->lines,  unsigned char*, node->alen * sizeof(unsigned char*), size * sizeof(unsigned char*));
+    GMEM_REALLOC(node->phases, unsigned char*, node->alen * sizeof(unsigned char*), size * sizeof(unsigned char*));
```

Comment:
I might be reading this wrong, but this seems to allocate both too much and too little memory at the same time.

It looks like `size` is the desired number of elements in `lines`, which are of type `unsigned char`. But instead of allocating `size * sizeof(unsigned char)` bytes, this allocates `size * sizeof(unsigned char*)`, 8 times more.

On the other hand `phases` uses full byte per line, so needs to be bigger than `lines` while here they have the same size.